nemo/lightning/pytorch/trainer.py

```diff
@@ -13,3 +28,120 @@ def io_init(self, **kwargs) -> fdl.Config[Self]:
         cfg_kwargs = {k: deepcopy(v) for k, v in kwargs.items()}
 
         return fdl.Config(type(self), **cfg_kwargs)
+
+    def setup_nemo(
+        self,
+        resume: Resume,
+        exp_dir: Optional[str] = None,
```

Comment:
It feels like the amount of arguments is quite a lot, and since this will be called inside `llm.train` we would need to expose everything there. I am wondering if we should introduce an `Experiment` class or something where can co-locate all those args. What do you think?